app/assets/javascripts/discourse/app/components/activation-email-form.hbs

```diff
@@ -1,2 +1,11 @@
 <p>{{i18n "login.provide_new_email"}}</p>
-<Input @value={{this.email}} class="activate-new-email" />
\ No newline at end of file
+
+{{#if @updateNewEmail}}
+  <Input
+    @value={{@email}}
+    {{on "input" @updateNewEmail}}
+    class="activate-new-email"
+  />
+{{else}}
+  <Input @value={{@email}} class="activate-new-email" />
```

Comment:
I was not able to find a way to conditionally add a modifier, and we are rendering this component in another template that is not needing the `@updateNewEmail` action... so this is my workaround

Reply:
I'd suggest adding an action to the component JS like

```
@action
newEmailChanged(){
  this.args.updateNewEmail?.()
}
```